runtime/service.go

```diff
@@ -755,7 +756,11 @@ func (s *service) Shutdown(ctx context.Context, req *taskAPI.ShutdownRequest) (*
 	}
 
 	// Exit to avoid 'zombie' shim processes
-	defer os.Exit(0)
+	defer func() {
+		s.shutdown()
```

Comment:
Probably want to check to see if `shutdown` is not `nil` before calling.

Reply:
done